### DISCUSSION TITLE
Auth Rehydration

### DISCUSSION THREAD

**Discussion by dtrinh100** [nuxt/nuxt#23100] [Questions]
I have some HTML that renders based on auth state from Pinia, is there a way to work around the rehydration issue without using `ClientOnly?
`
Sample rendering code:

```
<div v-if="authStore.isAuth()">
Render some login content
</div>
<div v-else>
Some unauth content
</div>
```

Pinia code:

```
function isAuth() {
return useCookie("auth").value && user; 
}
```

**Comment by mathix420** [comment]
We would need more details to help you on this, can you provide a reproduction example?

From what I can deduct, you should not have any rehydration issue in SSR mode if the cookie is available in both client and sever-side (not `HttpOnly`).
Try to `console.log` the content of your cookie in your `isAuth` function to see it's value changing, you will then see why and when it's value is different in client and server side.
